src/modules/storage/LedgerStorage.ts

```diff
@@ -523,6 +535,110 @@ export class LedgerStorage extends Storages
+                        VALUES
+                            (?, ?, ?, ?, ?, ?, ?, ? )
+                        ON DUPLICATE KEY UPDATE 
+                                    address = VALUES(address) , 
+                                    tx_count = VALUES(tx_count),
+                                    total_received = total_received + VALUES(total_received),
+                                    total_sent = total_sent + VALUES(total_sent),
+                                    total_freeze = VALUES(total_freeze),
+                                    total_spendable = VALUES(total_spendable),
+                                    total_balance = VALUES(total_balance)
+
+                        `,
+                    [
+                        address,
+                        accountInfo.tx_count,
+                        received_amount.toString(),
+                        sent_amount.toString(),
+                        "0",
+                        accountInfo.total_freeze.toString(),
+                        accountInfo.total_spendable.toString(),
+                        accountInfo.total_balance.toString(),
+                    ])
+                    .then(() => {
+                        resolve();
+                    })
+                    .catch((err) => {
+                        reject(err);
+                    });
+
+                return resolve();
+
+            });
+        }
+        return new Promise<void>((resolve, reject) => {
+            (async () => {
+                let block_transactions = `SELECT
+                 T.block_height, T.tx_hash, SUM(IFNULL(O.amount,0)) as amount,
+                T.tx_fee, T.tx_size, B.time_stamp,
+                JSON_ARRAYAGG(JSON_OBJECT("type", O.type, "address", O.address, "amount", O.amount)) as receiver,
+                (SELECT
+                   JSON_ARRAYAGG(JSON_OBJECT("type", O.type, "address", S.address, "amount", S.amount))
+                FROM
+                    blocks B
+                    INNER JOIN transactions T ON (B.height = T.block_height)
+                    INNER JOIN tx_inputs I ON (T.tx_hash = I.tx_hash)
+                    INNER JOIN tx_outputs S ON (I.utxo = S.utxo_key)
+                WHERE
+                    B.height= ? ) as sender_address
+            FROM
+                tx_outputs O
+                INNER JOIN transactions T ON (T.tx_hash = O.tx_hash)
+                INNER JOIN blocks B ON  (B.height = T.block_height)
+            WHERE
+                B.height = ?
+            GROUP BY T.tx_hash
+            ORDER BY T.tx_index ASC;`;
+
+                this.query(block_transactions, [block.header.height.value.toString(), block.header.height.value.toString()])
+                    .then(async (rows: any[]) => {
+                        let receivers = [];
+                        let senders = [];
+                        for (let tx_index = 0; tx_index < rows.length; tx_index++) {
```

Comment:
Make sure it's the same `rows.length` as the last `tx_index`., and leave an error message if it's wrong.